optimum/habana/diffusers/pipelines/cogvideox/pipeline_cogvideox.py

```diff
@@ -0,0 +1,458 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+import math
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+from diffusers import CogVideoXPipeline
+from diffusers.callbacks import MultiPipelineCallbacks, PipelineCallback
+from diffusers.models import AutoencoderKLCogVideoX, CogVideoXTransformer3DModel
+from diffusers.models.attention import Attention
+from diffusers.models.autoencoders.autoencoder_kl_cogvideox import CogVideoXCausalConv3d
+from diffusers.models.modeling_outputs import Transformer2DModelOutput
+from diffusers.pipelines.cogvideo.pipeline_output import CogVideoXPipelineOutput
+from diffusers.pipelines.cogvideo.pipeline_cogvideox import retrieve_timesteps
+from diffusers.schedulers import CogVideoXDDIMScheduler, CogVideoXDPMScheduler
+from diffusers.utils import (
+    USE_PEFT_BACKEND,
+    BaseOutput,
+    is_torch_version,
+    logging,
+    scale_lora_layers,
+    unscale_lora_layers,
+)
+from diffusers.utils.torch_utils import randn_tensor
+from transformers import T5EncoderModel, T5Tokenizer
+
+from optimum.habana.diffusers.models.attention_processor import CogVideoXAttnProcessorGaudi
+from optimum.habana.diffusers.models.cogvideox_transformer_3d import cogvideoXTransformerForwardGaudi
+from optimum.habana.diffusers.models.autoencoders.autoencoder_kl_cogvideox import tiled_decode_gaudi, CogVideoXCausalConv3dforwardGaudi
+from optimum.habana.diffusers.pipelines.pipeline_utils import GaudiDiffusionPipeline
+from optimum.habana.transformers.gaudi_configuration import GaudiConfig
```

Comment:
```suggestion
from ...models.attention_processor import CogVideoXAttnProcessorGaudi
from ...models.cogvideox_transformer_3d import cogvideoXTransformerForwardGaudi
from ...models.autoencoders.autoencoder_kl_cogvideox import tiled_decode_gaudi, CogVideoXCausalConv3dforwardGaudi
from ..pipeline_utils import GaudiDiffusionPipeline
from ....transformers.gaudi_configuration import GaudiConfig
```
Let's use relative imports to stay consistent with the rest of the codebase. Absolute imports  can sometimes lead to weird behaviours when two versions of the lib are installed in two different places.

Reply:
resolved